packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.tsx

```diff
@@ -801,6 +783,45 @@ export class ContextualMenu extends BaseComponent<IContextualMenuProps, IContext
     }
   }
 
+  /**
+   * Hnadles updating focus when mouseEnter or mouseMove fire.
+   * As part of updating focus, This function will also update
+   * the expand/collapse state accordingly.
+   */
+  private _updateFocusOnMouseEvent(item: any, ev: React.MouseEvent<HTMLElement>) {
+    const targetElement = ev.currentTarget as HTMLElement;
+
+    if (item.key === this.state.expandedMenuItemKey) {
+      return;
+    }
+
+    if (this._enterTimerId !== undefined) {
+      this._async.clearTimeout(this._enterTimerId);
+      this._enterTimerId = undefined;
+    }
+
+    const menuExpanded = this.state.expandedMenuItemKey !== undefined;
+
+    // If the menu is not expanded we can update focus without any delay
+    if (!menuExpanded) {
+      targetElement.focus();
```

Comment:
This will focus the item multiple times as the mouse moves through an item if the menu is not expanded.

Reply:
Nope because we won't call this code if we are already the activeElement:

`if (!this._isScrollIdle || targetElement === this._targetWindow.document.activeElement as HTMLElement) {
      return;
    }`